File: javascript/leetcode/Counting Elements.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var countElements = function (arr) {
  let count = 0;
  if (arr.length === 1) return count;

  const seen = new Set(arr);

  for (let el of arr) {
    if (seen.has(el + 1)) count++;
  }

  return count;
};

File: c++/leetcode/Counting Elements.cpp
class Solution
{
public:
  int countElements(vector<int> &arr)
  {
    int count = 0;
    if (arr.size() == 1)
      return count;

    set<int> seen{arr.begin(), arr.end()};

    for (auto el : arr)
    {
      if (seen.find(el + 1) != seen.end())
        count++;
    }

    return count;
  }
};